llvm/test/CodeGen/AMDGPU/GlobalISel/call-outgoing-stack-args.ll

```diff
@@ -223,9 +223,9 @@ define amdgpu_kernel void @kernel_caller_byval() {
 define void @func_caller_stack() {
 ; MUBUF-LABEL: func_caller_stack:
 ; MUBUF:       ; %bb.0:
-; MUBUF-NEXT:    s_waitcnt vmcnt(0) expcnt(0) lgkmcnt(0)
 ; MUBUF-NEXT:    s_mov_b32 s4, s33
 ; MUBUF-NEXT:    s_mov_b32 s33, s32
```

Comment:
This is assuming that SP/FP are never written by a memory instruction, which is probably an OK assumption but we should probably document it

Reply:
I don't want to change the ABI in this patch so I have "fixed" this by checking for reserved SGPRs and marking them as possibly depending on any of the counters.